### ISSUE TITLE
Self-designed model

### ISSUE THREAD

**Issue by AIUSR-TMP** [allenai/RL4LMs#13]
Hi there, I have a simple question. I want to know whether I can use this tool to train a self-designed model which cannot be found in huggingface. The input of our model is different from regular LM.

**Comment by rajcscw** [comment]
yes sure, you can definitely do this. But needs custom implementation of the [policy](https://github.com/allenai/RL4LMs/blob/2b54a97e3d7e295b2a119fada08153ef932b3e7f/rl4lms/envs/text_generation/policy/base_policy.py#L102). As long as you stick with the API of forward() methods, things should work. 

May be some tweaks for [trainer](https://github.com/allenai/RL4LMs/blob/2b54a97e3d7e295b2a119fada08153ef932b3e7f/rl4lms/envs/text_generation/training_utils.py#L126) might be needed.

**Comment by JulesGM** [comment]
this issue can likely be closed
